### ISSUE TITLE
[jsx] Automatic HTML indentation and cursor position in JSX

### ISSUE THREAD

**Issue by rickitan** [Microsoft/vscode#18284]
- VSCode Version:1.8.1 
- OS Version:10.12.2

VS code is not properly indenting and moving the cursor when writing HTML elements in JSX:

![vs-code-jsx-format](https://cloud.githubusercontent.com/assets/3137417/21754500/c46ce8da-d5cf-11e6-98cc-8bb7e7cb5111.gif)

Atom does it correctly:

![atom-jsx-format](https://cloud.githubusercontent.com/assets/3137417/21754505/d64a5b6e-d5cf-11e6-95de-2d26e69f8672.gif)

Does anybody know an extension or setting to fix it? I've been researching and I haven't been able to find a solution.

**Comment by rickitan** [comment]
Hi @aeschli,

I downloaded the night build and verified the the issue was fixed: 

![issue-fix-confirmed](https://cloud.githubusercontent.com/assets/3137417/21854129/96419bee-d7e7-11e6-9560-4eeae603bdf9.gif)

But I noticed a new issue in the night build that wasn't happening on the current stable release.

Autocomplete is not working on the night release:
![new-autocomplete-issue](https://cloud.githubusercontent.com/assets/3137417/21854150/ad01fa72-d7e7-11e6-872a-883ad66edabc.gif)

On stable it is: 

![autocomplete-working](https://cloud.githubusercontent.com/assets/3137417/21854163/c00fe6ce-d7e7-11e6-9871-41c1c3adbeb5.gif)

I got the same behavior in Javascript and Javascriptreact files. 

Should I open a new issue to report this? 

Thanks!
